consent-ws/src/main/java/org/broadinstitute/consent/http/service/DatabaseConsentAPI.java

```diff
@@ -230,30 +230,52 @@ public void logicalDelete(String id) throws UnknownIdentifierException {
         return getAllAssociationsForConsent(consentId);
     }
 
-    /** Can't add this to the DAO interface =( **/
-    private List<String> updateAssociations(String consentId, String associationType, List<String> ids){
+    /**
+     * Can't add this to the DAO interface =(
+     **/
+    private List<String> updateAssociations(String consentId, String associationType, List<String> ids) {
         Handle h = jdbi.open();
-        PreparedBatch insertBatch = h.prepareBatch("insert into consentassociations (consentId, associationType, objectId) values (?, ?, ?)");
-        for(String id: ids){
-            insertBatch.add(consentId, associationType, id);
+        PreparedBatch insertBatch = h.prepareBatch("insert into consentassociations (consentId, associationType, dataSetId) values (?, ?, ?)");
+        for (String id : ids) {
+            insertBatch.add(consentId, associationType, dataSetDAO.findDataSetIdByObjectId(id));
         }
         List<Long> insertedIds = insertBatch.executeAndGenerateKeys(LongColumnMapper.PRIMITIVE).list();
         h.close();
         List<String> stringsList = new ArrayList<>();
-        for (Long id: insertedIds) stringsList.add(id.toString());
+        for (Long id : insertedIds) stringsList.add(id.toString());
```

Comment:
Could be streamed a little more efficiently:
```suggestion
        List<String> stringsList = insertedIds.
            stream().
            map(String::valueOf).
            collect(Collectors.toList());
```